libsql-server/src/auth/user_auth_strategies/mod.rs

```diff
@@ -2,17 +2,26 @@ pub mod disabled;
 pub mod http_basic;
 pub mod jwt;
 
-use axum::http::HeaderValue;
+use anyhow::Context;
 pub use disabled::*;
 pub use http_basic::*;
 pub use jwt::*;
 
 use super::{AuthError, Authenticated};
 
 pub struct UserAuthContext {
-    pub user_credential: Option<HeaderValue>,
+    pub scheme: Option<String>,
+    pub token: Option<String>, // token might not be required in some cases
 }
 
+impl TryFrom<&str> for UserAuthContext {
+    type Error = anyhow::Error;
```

Comment:
I think we should create some errors via `thiserror` and use them in this module, we could do a sub error enum that then embeds in the overall server error enum. That way instead of using context + strings we qualify it via a typed enum.

Reply:
how about I add more variants into AuthError in auth/errors.rs ?

Reply:
done